sources/packages/backend/apps/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -23,16 +27,22 @@ import {
  */
 @Injectable()
 export class DesignationAgreementService extends RecordDataModelService<DesignationAgreement> {
-  constructor(private readonly dataSource: DataSource) {
+  private readonly institutionRepo: Repository<Institution>;
```

Comment:
We hardly declare the repositories following this approach.
Please check if the `InstitutionService` has a method already similar to the one needed. 
If not, create a method in the service and consume the service.